File: js/WebGL.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
import Common_1 from "./Common.js";
import Output_1 from "./Output.js";
import Mouse_1 from "./Mouse.js";
var WebGL = /** @class */ (function () {
    function WebGL(_a) {
        var $wrapper = _a.$wrapper;
        this.$wrapper = $wrapper;
        Common_1.init();
        Mouse_1.init();
        // this.init();
        this.$wrapper.prepend(Common_1.renderer.domElement);
        this.output = new Output_1();
        this.loop();
        window.addEventListener("resize", this.resize.bind(this));
    }
    WebGL.prototype.resize = function () {
        Common_1.resize();
        this.output.resize();
    };
    WebGL.prototype.render = function () {
        Mouse_1.update();
        Common_1.update();
        this.output.update();
    };
    WebGL.prototype.loop = function () {
        this.render();
        requestAnimationFrame(this.loop.bind(this));
    };
    return WebGL;
}());
const _default = WebGL;
export { _default as default };

File: lua/WebGL.lua
local window = js.global

local Common = require("lua.Common")
local Output = require("lua.Output")
local Mouse = require("lua.Mouse")

local WebGL = {}
local WebGLMT = {__index = WebGL}

function WebGL:new(_a)
    local self = {}

    self.wrapper = _a.wrapper
    Common:init()
    Mouse:init()
    self.wrapper:prepend(Common.renderer.DOMElement)
    self.output = Output:new()
    self:loop()
    window:addEventListener("resize", function ()
        self:resize()
    end)

    return setmetatable(self, WebGLMT)
end

function WebGL:resize()
    Common:resize()
    self.output:resize()
end

function WebGL:render()
    Mouse:update()
    Common:update()
    self.output:update()
end

function WebGL:loop()
    self:render()
    window:requestAnimationFrame(self:loop())
end

print("WebGL.lua initialized")
return WebGL